File: Era.java
import java.util.Scanner;
import java.lang.management.ManagementFactory;
import java.lang.management.ThreadMXBean;

public class Era {

    public static int sieve(int n) {
        boolean[] isComposite = new boolean[n + 1];

        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (!isComposite[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isComposite[j] = true;
                }
            }
        }

        int count = 0;
        for (int i = 2; i <= n; i++) {
            if (!isComposite[i]) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = sc.nextInt();

        // Performance measurement setup
        Runtime runtime = Runtime.getRuntime();
        ThreadMXBean bean = ManagementFactory.getThreadMXBean();

        runtime.gc(); // clean memory before measurement

        long memoryBefore = runtime.totalMemory() - runtime.freeMemory();
        long cpuBefore = bean.isCurrentThreadCpuTimeSupported()
                ? bean.getCurrentThreadCpuTime()
                : 0;
        long startTime = System.nanoTime();

        int result = sieve(n);

        long endTime = System.nanoTime();
        long cpuAfter = bean.isCurrentThreadCpuTimeSupported()
                ? bean.getCurrentThreadCpuTime()
                : 0;
        long memoryAfter = runtime.totalMemory() - runtime.freeMemory();

        double wallClockSeconds = (endTime - startTime) / 1_000_000_000.0;
        double cpuSeconds = (cpuAfter - cpuBefore) / 1_000_000_000.0;
        long memoryUsedBytes = memoryAfter - memoryBefore;

        System.out.println("\nNumber of primes ≤ " + n + ": " + result);
        System.out.printf("Wall-clock time: %.6f seconds%n", wallClockSeconds);
        System.out.printf("CPU time: %.6f seconds%n", cpuSeconds);
        System.out.printf("Memory change: %.2f KB%n", memoryUsedBytes / 1024.0);

        sc.close();
    }
}

File: era.py
import time
import os
import psutil
import math

def sieve(n):
    is_composite = [False] * (n + 1)
    for i in range(2, int(math.sqrt(n)) + 1):
        if not is_composite[i]:
            for j in range(i * i, n + 1, i):
                is_composite[j] = True
    count = sum(1 for i in range(2, n + 1) if not is_composite[i])
    return count

if __name__ == "__main__":
    n = int(input("Enter a number: "))

    process = psutil.Process(os.getpid())

    # Clean up memory before measurement
    process.memory_info()  # Force update
    mem_before = process.memory_info().rss  # in bytes
    cpu_times_before = process.cpu_times()
    start_time = time.perf_counter()

    result = sieve(n)

    end_time = time.perf_counter()
    cpu_times_after = process.cpu_times()
    mem_after = process.memory_info().rss  # in bytes

    wall_clock_seconds = end_time - start_time
    cpu_seconds = (cpu_times_after.user - cpu_times_before.user) + \
                  (cpu_times_after.system - cpu_times_before.system)
    memory_change_kb = (mem_after - mem_before) / 1024.0

    print(f"\nNumber of primes ≤ {n}: {result}")
    print(f"Wall-clock time: {wall_clock_seconds:.6f} seconds")
    print(f"CPU time: {cpu_seconds:.6f} seconds")
    print(f"Memory change: {memory_change_kb:.2f} KB")
